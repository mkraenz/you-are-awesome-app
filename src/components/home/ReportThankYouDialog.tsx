import React, { FC } from "react";
import { useTranslation } from "react-i18next";
import { Button, Dialog, Paragraph, Portal } from "react-native-paper";

interface Props {
    onDismiss: () => void;
}

const ReportThankYouDialog: FC<Props> = ({ onDismiss }) => {
    const { t } = useTranslation();
    return (
        <Portal>
            <Dialog visible={true} onDismiss={onDismiss}>
                <Dialog.Title accessibilityStates={{}}>
                    {t("reportThankYouTitle")}
                </Dialog.Title>
                <Dialog.Content>
                    <Paragraph>{t("reportThankYouText")}</Paragraph>
                </Dialog.Content>
                <Dialog.Actions>
                    <Button onPress={onDismiss} accessibilityStates={{}}>
                        {t(`reportThankYouButton`)}
                    </Button>
                </Dialog.Actions>
            </Dialog>
        </Portal>
    );
};
export default ReportThankYouDialog;
