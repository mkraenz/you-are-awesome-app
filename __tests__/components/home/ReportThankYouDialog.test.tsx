import i18next from "i18next";
import { noop } from "lodash";
import React from "react";
import "react-native";
import renderer, { act } from "react-test-renderer";
import ReportThankYouDialog from "../../../src/components/home/ReportThankYouDialog";
import TestLocalizationProvider from "../../helpers/TestLocalizationProvider";

it("renders correctly", async () => {
    const tree = renderer
        .create(
            <TestLocalizationProvider>
                <ReportThankYouDialog onDismiss={noop} />
            </TestLocalizationProvider>
        )
        .toJSON();
    await act(async () => {});

    const treeStr = JSON.stringify(tree);
    expect(treeStr).toContain(i18next.t("reportThankYouTitle"));
    expect(treeStr).toContain(i18next.t("reportThankYouText"));
    expect(treeStr).toContain(i18next.t("reportThankYouButton"));
    expect(tree).toMatchSnapshot();
});
