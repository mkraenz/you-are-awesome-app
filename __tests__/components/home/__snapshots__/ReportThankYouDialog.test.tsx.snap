// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityLiveRegion="polite"
  accessibilityViewIsModal={true}
  collapsable={false}
  onAccessibilityEscape={[Function]}
  pointerEvents="auto"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    accessibilityLabel="Close modal"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    importantForAccessibility="no"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "opacity": 1,
      }
    }
  />
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "marginBottom": 0,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 4,
          "elevation": 24,
          "justifyContent": "flex-start",
          "marginHorizontal": 26,
          "marginVertical": 0,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 23,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 24,
        }
      }
    >
      <Text
        accessibilityRole="header"
        accessibilityTraits="header"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "marginBottom": 18,
                    "marginHorizontal": 24,
                    "marginTop": 22,
                  },
                  Object {
                    "color": "#000000",
                  },
                  undefined,
                ],
              ],
            ],
          ]
        }
      >
        Thanks for Reporting
      </Text>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 24,
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 14,
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          We are committed to protecting our users and others as well as their rights. We will check your report as soon as possible and take appropriate action.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 0,
              "minWidth": "auto",
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      Object {
                        "marginHorizontal": 8,
                      },
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
